Refactor generator requests to support non-JS languages

RoboErikG opened the pull request:
This adds a LanguageDefinition that includes the path to the base
language and the object namespace that language uses and makes it
required for code generation. Adds a test for generating Lua.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/google/blockly-android/592)
<!-- Reviewable:end -->

RoboErikG commented on blocklylib-core/src/main/java/com/google/blockly/android/codegen/CodeGenerationRequest.java:
```diff
@@ -73,4 +81,37 @@ public String getXml() {
+     * Blockly needs to know the path to the file and the object namespace that has the generator
+     * functions. For example: {}"javascript_compressed.js", "Blockly.JavaScript"}
+     */
+    public static class LanguageDefinition {
```
Thoughts on whether this should get its own file?

AnmAtAnm commented:
Reviewed 4 of 30 files at r1.
Review status: 4 of 30 files reviewed at latest revision, 5 unresolved discussions, some commit checks failed.

---

*[blocklydemo/src/main/java/com/google/blockly/android/demo/LuaActivity.java, line 43 at r2](https://reviewable.io:443/reviews/google/blockly-android/592#-KjoUUT0zQTVi1MrVaM_:-KjoUUT0zQTVi1MrVaMa:bbh599c) ([raw file](https://github.com/google/blockly-android/blob/e28efd6657aae931c501a328a107a8997fe1c8d1/blocklydemo/src/main/java/com/google/blockly/android/demo/LuaActivity.java#L43)):*
> ```Java
>  */
> public class LuaActivity extends AbstractBlocklyActivity {
>     private static final String TAG = "SplitActivity";
> ```

wrong class

---

*[blocklydemo/src/main/java/com/google/blockly/android/demo/LuaActivity.java, line 47 at r2](https://reviewable.io:443/reviews/google/blockly-android/592#-KjoUjw_lYURIhAsgISi:-KjoUjw_lYURIhAsgISj:b-lk2070) ([raw file](https://github.com/google/blockly-android/blob/e28efd6657aae931c501a328a107a8997fe1c8d1/blocklydemo/src/main/java/com/google/blockly/android/demo/LuaActivity.java#L47)):*
> ```Java
>     private static final String SAVED_WORKSPACE_FILENAME = "lua_workspace.xml";
>     // Add custom blocks to this list.
>     private static final List<String> BLOCK_DEFINITIONS = Arrays.asList(
> ```

DefaultBlocks.ALL_BLOCK_DEFINITIONS

---

*[blocklydemo/src/main/java/com/google/blockly/android/demo/LuaActivity.java, line 114 at r2](https://reviewable.io:443/reviews/google/blockly-android/592#-KjoU1tau3qkqykmE5kH:-KjoU1tau3qkqykmE5kI:bbsz188) ([raw file](https://github.com/google/blockly-android/blob/e28efd6657aae931c501a328a107a8997fe1c8d1/blocklydemo/src/main/java/com/google/blockly/android/demo/LuaActivity.java#L114)):*
> ```Java
>     @Override
>     protected int getActionBarMenuResId() {
>         return R.menu.turtle_actionbar;
> ```

This has items for Turtle specific workspaces, like "Android", "Lacey Curves" and "Paint Strokes".

---

*[blocklylib-core/src/main/java/com/google/blockly/android/AbstractBlocklyActivity.java, line 397 at r2](https://reviewable.io:443/reviews/google/blockly-android/592#-KjoVK1aOOpFGvlShRyu:-KjoVK1aOOpFGvlShRyv:b-7b479b) ([raw file](https://github.com/google/blockly-android/blob/e28efd6657aae931c501a328a107a8997fe1c8d1/blocklylib-core/src/main/java/com/google/blockly/android/AbstractBlocklyActivity.java#L397)):*
> ```Java
>      */
>     @NonNull
>     abstract protected CodeGenerationRequest.LanguageDefinition getBlockGeneratorLanguage();
> ```

Let this default to the JavaScript language definition.
I don't see us wanting to ship the JavaScript assets separately, and that we are pushing JavaScript for the Google runtime.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly-android/592)*
<!-- Sent from Reviewable.io -->

AnmAtAnm commented:
Review status: 4 of 30 files reviewed at latest revision, 5 unresolved discussions, some commit checks failed.

---

*[blocklylib-core/src/main/java/com/google/blockly/android/codegen/CodeGenerationRequest.java, line 89 at r1](https://reviewable.io:443/reviews/google/blockly-android/592#-KjoPZwiBenCW6HvmaW3-r1-89:-KjoW34exXqQ-JvlAgL0:bbmrra7) ([raw file](https://github.com/google/blockly-android/blob/2602ffc88b3905d36cb292f00edf525228e5e2aa/blocklylib-core/src/main/java/com/google/blockly/android/codegen/CodeGenerationRequest.java#L89)):*
<details><summary><i>Previously, RoboErikG wrote…</i></summary><blockquote>

Thoughts on whether this should get its own file?
</blockquote></details>

Definitely think it deserves it's own file, probably with the JavaScript default nested within.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly-android/592)*
<!-- Sent from Reviewable.io -->

AnmAtAnm commented:
Reviewed 1 of 30 files at r1.
Review status: 5 of 30 files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---

*[blocklylib-core/src/main/java/com/google/blockly/android/codegen/CodeGenerationRequest.java, line 92 at r2](https://reviewable.io:443/reviews/google/blockly-android/592#-KjoWaA5yZfVmMquvfeX:-KjoWaA6sfsCs5zYUtx8:b9ofyc5) ([raw file](https://github.com/google/blockly-android/blob/e28efd6657aae931c501a328a107a8997fe1c8d1/blocklylib-core/src/main/java/com/google/blockly/android/codegen/CodeGenerationRequest.java#L92)):*
> ```Java
>         public final String mLanguageFilename;
>         public final String mLanguageNamespace;
> ```

This would be a great place to define reserved words in a way that the NameManagers can access them.
https://github.com/google/blockly/blob/master/generators/javascript.js#L45
https://github.com/google/blockly/blob/master/generators/lua.js#L46

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly-android/592)*
<!-- Sent from Reviewable.io -->

vicng commented:
Review status: 5 of 30 files reviewed at latest revision, 15 unresolved discussions, some commit checks failed.

---

*[blocklydemo/src/main/java/com/google/blockly/android/demo/LuaActivity.java, line 2 at r2](https://reviewable.io:443/reviews/google/blockly-android/592#-KjoYC7bH5aRkuHHPNWi:-KjoYC7bH5aRkuHHPNWj:bb7s9mq) ([raw file](https://github.com/google/blockly-android/blob/e28efd6657aae931c501a328a107a8997fe1c8d1/blocklydemo/src/main/java/com/google/blockly/android/demo/LuaActivity.java#L2)):*
> ```Java
> /*
>  *  Copyright 2015 Google Inc. All Rights Reserved.
> ```

year

---

*[blocklylib-core/src/main/assets/background_compiler.html, line 12 at r2](https://reviewable.io:443/reviews/google/blockly-android/592#-Kjo_UwpozrGK-fcmZcO:-Kjo_UwpozrGK-fcmZcP:b22lvlt) ([raw file](https://github.com/google/blockly-android/blob/e28efd6657aae931c501a328a107a8997fe1c8d1/blocklylib-core/src/main/assets/background_compiler.html#L12)):*
> ```HTML
>     <script>
>         function import_generator_language(filename) {
>             console.log('importing generator language ' + filename);
> ```

Remove log.

---

*[blocklylib-core/src/main/java/com/google/blockly/android/BlocklyActivityHelper.java, line 91 at r2](https://reviewable.io:443/reviews/google/blockly-android/592#-KjoaqR8uLUHBMLdHKC1:-KjoaqR8uLUHBMLdHKC2:b-qlbpqm) ([raw file](https://github.com/google/blockly-android/blob/e28efd6657aae931c501a328a107a8997fe1c8d1/blocklylib-core/src/main/java/com/google/blockly/android/BlocklyActivityHelper.java#L91)):*
> ```Java
>     protected BlocklyController mController;
>     protected CodeGeneratorManager mCodeGeneratorManager;
>     protected String mCodeGeneratorLanguage = "javascript_compressed.js";
> ```

Where's this used?

---

*[blocklylib-core/src/main/java/com/google/blockly/android/codegen/CodeGenerationRequest.java, line 87 at r2](https://reviewable.io:443/reviews/google/blockly-android/592#-KjobWt0Mgx1Fq6d6MOn:-KjobWt0Mgx1Fq6d6MOo:blfkrvd) ([raw file](https://github.com/google/blockly-android/blob/e28efd6657aae931c501a328a107a8997fe1c8d1/blocklylib-core/src/main/java/com/google/blockly/android/codegen/CodeGenerationRequest.java#L87)):*
> ```Java
>      * Defines the core language file to be used in code generation. To be used by the generator
>      * Blockly needs to know the path to the file and the object namespace that has the generator
>      * functions. For example: {}"javascript_compressed.js", "Blockly.JavaScript"}
> ```

typo {}

---

*[blocklylib-core/src/main/java/com/google/blockly/android/codegen/CodeGenerationRequest.java, line 97 at r2](https://reviewable.io:443/reviews/google/blockly-android/592#-KjobeHE0MQVPAIsvsNN:-KjobeHE0MQVPAIsvsNO:bitjbt) ([raw file](https://github.com/google/blockly-android/blob/e28efd6657aae931c501a328a107a8997fe1c8d1/blocklylib-core/src/main/java/com/google/blockly/android/codegen/CodeGenerationRequest.java#L97)):*
> ```Java
>          *
>          * @param filename The path to the language file relative to
>          *                 file:///android_assets/background_compiler.html.
> ```

This is the path to the generator file right?

---

*[blocklylib-core/src/main/java/com/google/blockly/android/codegen/CodeGeneratorService.java, line 216 at r2](https://reviewable.io:443/reviews/google/blockly-android/592#-KjoYmC4U5kgNDJYmejV:-KjoYmC4U5kgNDJYmejW:b-eef7e2) ([raw file](https://github.com/google/blockly-android/blob/e28efd6657aae931c501a328a107a8997fe1c8d1/blocklylib-core/src/main/java/com/google/blockly/android/codegen/CodeGeneratorService.java#L216)):*
> ```Java
>         @JavascriptInterface
>         public String getGeneratorLanguageFilename() {
>             if (mGeneratorLanguage == null) {
> ```

Seems strange that this defaults to javascript.

Either remove or set it on instantiation.

---

*[blocklylib-core/src/main/java/com/google/blockly/model/DefaultBlocks.java, line 43 at r2](https://reviewable.io:443/reviews/google/blockly-android/592#-KjoaBrnxMbyagb8HVRq:-KjoaBrnxMbyagb8HVRr:b-bqwal4) ([raw file](https://github.com/google/blockly-android/blob/e28efd6657aae931c501a328a107a8997fe1c8d1/blocklylib-core/src/main/java/com/google/blockly/model/DefaultBlocks.java#L43)):*
> ```Java
>      * Standard definition for the JavaScript language generator.
>      */
>     public static final CodeGenerationRequest.LanguageDefinition JAVASCRIPT_LANGUAGE_DEF
> ```

Shouldn't this be defined in LanguageDefinition?

---

*[blocklytest/src/androidTest/assets/lua/generators/test_blocks.js, line 22 at r2](https://reviewable.io:443/reviews/google/blockly-android/592#-Kjoc5S-tByvLk1XBDVN:-Kjoc5S-tByvLk1XBDVO:b-ikx6ew) ([raw file](https://github.com/google/blockly-android/blob/e28efd6657aae931c501a328a107a8997fe1c8d1/blocklytest/src/androidTest/assets/lua/generators/test_blocks.js#L22)):*
> ```JavaScript
> 
> /**
>  * @fileoverview Generating Lua for colour blocks.
> ```

Update comments.

---

*[blocklytest/src/androidTest/java/com/google/blockly/android/codegen/CodeGeneratorServiceTest.java, line 79 at r2](https://reviewable.io:443/reviews/google/blockly-android/592#-KjoZbiZwcWDvM3rBqP4:-KjoZbiZwcWDvM3rBqP5:b-fm6kyb) ([raw file](https://github.com/google/blockly-android/blob/e28efd6657aae931c501a328a107a8997fe1c8d1/blocklytest/src/androidTest/java/com/google/blockly/android/codegen/CodeGeneratorServiceTest.java#L79)):*
> ```Java
>                 Arrays.asList(new String[] {"lua/generators/test_blocks.js"}));
>         mManager.requestCodeGeneration(request);
>         Mockito.verify(mCallback, Mockito.timeout(500)).onFinishCodeGeneration("local _ = 'test'\n");
> ```

line length

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly-android/592)*
<!-- Sent from Reviewable.io -->

RoboErikG commented:
Review status: 3 of 21 files reviewed at latest revision, 15 unresolved discussions, some commit checks failed.

---

*[blocklydemo/src/main/java/com/google/blockly/android/demo/LuaActivity.java, line 2 at r2](https://reviewable.io:443/reviews/google/blockly-android/592#-KjoYC7bH5aRkuHHPNWi:-KjomXZVPERcbEkcX6Mk:b-896fix) ([raw file](https://github.com/google/blockly-android/blob/e28efd6657aae931c501a328a107a8997fe1c8d1/blocklydemo/src/main/java/com/google/blockly/android/demo/LuaActivity.java#L2)):*
<details><summary><i>Previously, vicng wrote…</i></summary><blockquote>

year
</blockquote></details>

Done.

---

*[blocklydemo/src/main/java/com/google/blockly/android/demo/LuaActivity.java, line 43 at r2](https://reviewable.io:443/reviews/google/blockly-android/592#-KjoUUT0zQTVi1MrVaM_:-KjoZGn7tIYzzGn41Sjy:b-896fix) ([raw file](https://github.com/google/blockly-android/blob/e28efd6657aae931c501a328a107a8997fe1c8d1/blocklydemo/src/main/java/com/google/blockly/android/demo/LuaActivity.java#L43)):*
<details><summary><i>Previously, AnmAtAnm (Andrew n marshall) wrote…</i></summary><blockquote>

wrong class
</blockquote></details>

Done.

---

*[blocklydemo/src/main/java/com/google/blockly/android/demo/LuaActivity.java, line 47 at r2](https://reviewable.io:443/reviews/google/blockly-android/592#-KjoUjw_lYURIhAsgISi:-KjoZDnY2Y-X5u_xuH7X:b-896fix) ([raw file](https://github.com/google/blockly-android/blob/e28efd6657aae931c501a328a107a8997fe1c8d1/blocklydemo/src/main/java/com/google/blockly/android/demo/LuaActivity.java#L47)):*
<details><summary><i>Previously, AnmAtAnm (Andrew n marshall) wrote…</i></summary><blockquote>

DefaultBlocks.ALL_BLOCK_DEFINITIONS
</blockquote></details>

Done.

---

*[blocklydemo/src/main/java/com/google/blockly/android/demo/LuaActivity.java, line 114 at r2](https://reviewable.io:443/reviews/google/blockly-android/592#-KjoU1tau3qkqykmE5kH:-KjoZgDaxcPO-k_HzRM6:b-896fix) ([raw file](https://github.com/google/blockly-android/blob/e28efd6657aae931c501a328a107a8997fe1c8d1/blocklydemo/src/main/java/com/google/blockly/android/demo/LuaActivity.java#L114)):*
<details><summary><i>Previously, AnmAtAnm (Andrew n marshall) wrote…</i></summary><blockquote>

This has items for Turtle specific workspaces, like "Android", "Lacey Curves" and "Paint Strokes".
</blockquote></details>

Done.

---

*[blocklylib-core/src/main/assets/background_compiler.html, line 12 at r2](https://reviewable.io:443/reviews/google/blockly-android/592#-Kjo_UwpozrGK-fcmZcO:-Kjombp-sVPEAWsK0BPn:b-896fix) ([raw file](https://github.com/google/blockly-android/blob/e28efd6657aae931c501a328a107a8997fe1c8d1/blocklylib-core/src/main/assets/background_compiler.html#L12)):*
<details><summary><i>Previously, vicng wrote…</i></summary><blockquote>

Remove log.
</blockquote></details>

Done.

---

*[blocklylib-core/src/main/java/com/google/blockly/android/AbstractBlocklyActivity.java, line 397 at r2](https://reviewable.io:443/reviews/google/blockly-android/592#-KjoVK1aOOpFGvlShRyu:-KjoZqxai4aulHKyK0XD:b-896fix) ([raw file](https://github.com/google/blockly-android/blob/e28efd6657aae931c501a328a107a8997fe1c8d1/blocklylib-core/src/main/java/com/google/blockly/android/AbstractBlocklyActivity.java#L397)):*
<details><summary><i>Previously, AnmAtAnm (Andrew n marshall) wrote…</i></summary><blockquote>

Let this default to the JavaScript language definition.
I don't see us wanting to ship the JavaScript assets separately, and that we are pushing JavaScript for the Google runtime.
</blockquote></details>

Done.

---

*[blocklylib-core/src/main/java/com/google/blockly/android/BlocklyActivityHelper.java, line 91 at r2](https://reviewable.io:443/reviews/google/blockly-android/592#-KjoaqR8uLUHBMLdHKC1:-KjomhlbCLg8wNqu79LH:b-896fix) ([raw file](https://github.com/google/blockly-android/blob/e28efd6657aae931c501a328a107a8997fe1c8d1/blocklylib-core/src/main/java/com/google/blockly/android/BlocklyActivityHelper.java#L91)):*
<details><summary><i>Previously, vicng wrote…</i></summary><blockquote>

Where's this used?
</blockquote></details>

Done.

---

*[blocklylib-core/src/main/java/com/google/blockly/android/codegen/CodeGenerationRequest.java, line 89 at r1](https://reviewable.io:443/reviews/google/blockly-android/592#-KjoPZwiBenCW6HvmaW3-r1-89:-KjomqUoHt83iP9ztFP0:b-896fix) ([raw file](https://github.com/google/blockly-android/blob/2602ffc88b3905d36cb292f00edf525228e5e2aa/blocklylib-core/src/main/java/com/google/blockly/android/codegen/CodeGenerationRequest.java#L89)):*
<details><summary><i>Previously, AnmAtAnm (Andrew n marshall) wrote…</i></summary><blockquote>

Definitely think it deserves it's own file, probably with the JavaScript default nested within.
</blockquote></details>

Done.

---

*[blocklylib-core/src/main/java/com/google/blockly/android/codegen/CodeGenerationRequest.java, line 87 at r2](https://reviewable.io:443/reviews/google/blockly-android/592#-KjobWt0Mgx1Fq6d6MOn:-Kjompe01HuKsDbb31aY:b-896fix) ([raw file](https://github.com/google/blockly-android/blob/e28efd6657aae931c501a328a107a8997fe1c8d1/blocklylib-core/src/main/java/com/google/blockly/android/codegen/CodeGenerationRequest.java#L87)):*
<details><summary><i>Previously, vicng wrote…</i></summary><blockquote>

typo {}
</blockquote></details>

Done.

---

*[blocklylib-core/src/main/java/com/google/blockly/android/codegen/CodeGenerationRequest.java, line 92 at r2](https://reviewable.io:443/reviews/google/blockly-android/592#-KjoWaA5yZfVmMquvfeX:-KjokPn19bbgRx6u0trA:bkd5ega) ([raw file](https://github.com/google/blockly-android/blob/e28efd6657aae931c501a328a107a8997fe1c8d1/blocklylib-core/src/main/java/com/google/blockly/android/codegen/CodeGenerationRequest.java#L92)):*
<details><summary><i>Previously, AnmAtAnm (Andrew n marshall) wrote…</i></summary><blockquote>

This would be a great place to define reserved words in a way that the NameManagers can access them.
https://github.com/google/blockly/blob/master/generators/javascript.js#L45
https://github.com/google/blockly/blob/master/generators/lua.js#L46
</blockquote></details>

Added a TODO

---

*[blocklylib-core/src/main/java/com/google/blockly/android/codegen/CodeGenerationRequest.java, line 97 at r2](https://reviewable.io:443/reviews/google/blockly-android/592#-KjobeHE0MQVPAIsvsNN:-KjokRewBI61oMg7G7yd:btqqu5l) ([raw file](https://github.com/google/blockly-android/blob/e28efd6657aae931c501a328a107a8997fe1c8d1/blocklylib-core/src/main/java/com/google/blockly/android/codegen/CodeGenerationRequest.java#L97)):*
<details><summary><i>Previously, vicng wrote…</i></summary><blockquote>

This is the path to the generator file right?
</blockquote></details>

Correct.

---

*[blocklylib-core/src/main/java/com/google/blockly/android/codegen/CodeGeneratorService.java, line 216 at r2](https://reviewable.io:443/reviews/google/blockly-android/592#-KjoYmC4U5kgNDJYmejV:-KjolGsNIVzdjrC2-taK:b-p5u6wm) ([raw file](https://github.com/google/blockly-android/blob/e28efd6657aae931c501a328a107a8997fe1c8d1/blocklylib-core/src/main/java/com/google/blockly/android/codegen/CodeGeneratorService.java#L216)):*
<details><summary><i>Previously, vicng wrote…</i></summary><blockquote>

Seems strange that this defaults to javascript.

Either remove or set it on instantiation.
</blockquote></details>

added some checks in the request to require it and made this throw if it's missing.

---

*[blocklylib-core/src/main/java/com/google/blockly/model/DefaultBlocks.java, line 43 at r2](https://reviewable.io:443/reviews/google/blockly-android/592#-KjoaBrnxMbyagb8HVRq:-KjomCLKPUCG4Vpox9HF:b-896fix) ([raw file](https://github.com/google/blockly-android/blob/e28efd6657aae931c501a328a107a8997fe1c8d1/blocklylib-core/src/main/java/com/google/blockly/model/DefaultBlocks.java#L43)):*
<details><summary><i>Previously, vicng wrote…</i></summary><blockquote>

Shouldn't this be defined in LanguageDefinition?
</blockquote></details>

Done.

---

*[blocklytest/src/androidTest/assets/lua/generators/test_blocks.js, line 22 at r2](https://reviewable.io:443/reviews/google/blockly-android/592#-Kjoc5S-tByvLk1XBDVN:-KjomMAk6ZajBwatLXJ4:b-896fix) ([raw file](https://github.com/google/blockly-android/blob/e28efd6657aae931c501a328a107a8997fe1c8d1/blocklytest/src/androidTest/assets/lua/generators/test_blocks.js#L22)):*
<details><summary><i>Previously, vicng wrote…</i></summary><blockquote>

Update comments.
</blockquote></details>

Done.

---

*[blocklytest/src/androidTest/java/com/google/blockly/android/codegen/CodeGeneratorServiceTest.java, line 79 at r2](https://reviewable.io:443/reviews/google/blockly-android/592#-KjoZbiZwcWDvM3rBqP4:-KjomS4obn2AB_KH-EqQ:b-896fix) ([raw file](https://github.com/google/blockly-android/blob/e28efd6657aae931c501a328a107a8997fe1c8d1/blocklytest/src/androidTest/java/com/google/blockly/android/codegen/CodeGeneratorServiceTest.java#L79)):*
<details><summary><i>Previously, vicng wrote…</i></summary><blockquote>

line length
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly-android/592)*
<!-- Sent from Reviewable.io -->

vicng commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 3 of 21 files reviewed at latest revision, 15 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly-android/592#-:-Kjor-3Q-RMren4JpaYR:bnfp4nl)*
<!-- Sent from Reviewable.io -->

amalcao commented on blocklylib-core/src/main/assets/background_compiler.html:
```diff
@@ -40,14 +48,15 @@
         }
 
         import_generators(BlocklyJavascriptInterface.getBlockGeneratorsFilenames());
+        import_generator_language(BlocklyJavascriptInterface.getGeneratorLanguageFilename());
```
Why not import the default language specific generating script before the user defined ones?
Maybe someone wants to override the functions defined in the default generator.

RoboErikG replied:
Whoops, good catch, Amal. Fixed.

AnmAtAnm commented:
Reviewed 2 of 30 files at r1, 14 of 27 files at r3, 1 of 2 files at r4, 1 of 1 files at r6.
Review status: 20 of 21 files reviewed at latest revision, 17 unresolved discussions, some commit checks failed.

---

*[blocklydemo/src/main/java/com/google/blockly/android/demo/SplitActivity.java, line 30 at r5](https://reviewable.io:443/reviews/google/blockly-android/592#-KjseOFfUu_7pFv7jGgs:-KjseOFgfXW3apPnHWsN:buzc58k) ([raw file](https://github.com/google/blockly-android/blob/fe40f53f19087814bb0a3f0ddf206e89cbc733b0/blocklydemo/src/main/java/com/google/blockly/android/demo/SplitActivity.java#L30)):*
> ```Java
> import com.google.blockly.android.codegen.CodeGenerationRequest;
> import com.google.blockly.model.BlocklySerializerException;
> import com.google.blockly.model.DefaultBlocks;
> ```

unused?

---

*[blocklydemo/src/main/java/com/google/blockly/android/demo/StylesActivity.java, line 22 at r5](https://reviewable.io:443/reviews/google/blockly-android/592#-KjsehoXn58OjPMK6sfR:-KjsehoXn58OjPMK6sfS:buzc58k) ([raw file](https://github.com/google/blockly-android/blob/fe40f53f19087814bb0a3f0ddf206e89cbc733b0/blocklydemo/src/main/java/com/google/blockly/android/demo/StylesActivity.java#L22)):*
> ```Java
> import com.google.blockly.android.codegen.CodeGenerationRequest;
> import com.google.blockly.android.codegen.LoggingCodeGeneratorCallback;
> import com.google.blockly.model.DefaultBlocks;
> ```

unused?

---

*[blocklydemo/src/main/java/com/google/blockly/android/demo/flyout/AlwaysOpenToolboxActivity.java, line 24 at r5](https://reviewable.io:443/reviews/google/blockly-android/592#-Kjsf0ljrbAeuf2PPvY8:-Kjsf0lkUvIgY_kv1w_D:b-ciom1o) ([raw file](https://github.com/google/blockly-android/blob/fe40f53f19087814bb0a3f0ddf206e89cbc733b0/blocklydemo/src/main/java/com/google/blockly/android/demo/flyout/AlwaysOpenToolboxActivity.java#L24)):*
> ```Java
> import com.google.blockly.android.codegen.LoggingCodeGeneratorCallback;
> import com.google.blockly.android.demo.R;
> import com.google.blockly.model.DefaultBlocks;
> ```

Unused?

---

*[blocklydemo/src/main/java/com/google/blockly/android/demo/flyout/NoCategoriesToolboxActivity.java, line 11 at r5](https://reviewable.io:443/reviews/google/blockly-android/592#-Kjsf2WqOyOnghc_vi79:-Kjsf2WrS8bngvXWAhC1:b-ciom1o) ([raw file](https://github.com/google/blockly-android/blob/fe40f53f19087814bb0a3f0ddf206e89cbc733b0/blocklydemo/src/main/java/com/google/blockly/android/demo/flyout/NoCategoriesToolboxActivity.java#L11)):*
> ```Java
> import com.google.blockly.android.codegen.LoggingCodeGeneratorCallback;
> import com.google.blockly.android.demo.R;
> import com.google.blockly.model.DefaultBlocks;
> ```

Unused?

---

*[blocklylib-core/src/main/java/com/google/blockly/android/codegen/CodeGenerationRequest.java, line 44 at r6](https://reviewable.io:443/reviews/google/blockly-android/592#-KjshBmCxlXlbyQwTVcl:-KjshBmCxlXlbyQwTVcm:b-iaqngo) ([raw file](https://github.com/google/blockly-android/blob/739ce02273f11a19cf64e41a148b103c2a03427e/blocklylib-core/src/main/java/com/google/blockly/android/codegen/CodeGenerationRequest.java#L44)):*
> ```Java
>      */
>     public CodeGenerationRequest(String xml, CodeGeneratorCallback callback,
>             LanguageDefinition generatorsLanguage, List<String> blockDefinitionsFilenames,
> ```

Might as well mark these params @NotNull.

---

*[blocklylib-core/src/main/java/com/google/blockly/android/codegen/LanguageDefinition.java, line 17 at r6](https://reviewable.io:443/reviews/google/blockly-android/592#-KjshzfZyTJ5lpU2c49J:-KjshzfZyTJ5lpU2c49K:bzga3rz) ([raw file](https://github.com/google/blockly-android/blob/739ce02273f11a19cf64e41a148b103c2a03427e/blocklylib-core/src/main/java/com/google/blockly/android/codegen/LanguageDefinition.java#L17)):*
> ```Java
> 
>     public final String mLanguageFilename;
>     public final String mLanguageNamespace;
> ```

JavaDoc these public fields.
Also, mLanguageNamespace => mGeneratorRef, per offline.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly-android/592)*
<!-- Sent from Reviewable.io -->

AnmAtAnm commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> with nits.

---

Review status: 20 of 21 files reviewed at latest revision, 17 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly-android/592#-:-KjsnMUZZHomVpq0dn5-:b-cwzrcn)*
<!-- Sent from Reviewable.io -->

RoboErikG commented:
Review status: 14 of 18 files reviewed at latest revision, 16 unresolved discussions, some commit checks failed.

---

*[blocklydemo/src/main/java/com/google/blockly/android/demo/SplitActivity.java, line 30 at r5](https://reviewable.io:443/reviews/google/blockly-android/592#-KjseOFfUu_7pFv7jGgs:-Kjt002Y3q6fE7bJGMSc:b-896fix) ([raw file](https://github.com/google/blockly-android/blob/fe40f53f19087814bb0a3f0ddf206e89cbc733b0/blocklydemo/src/main/java/com/google/blockly/android/demo/SplitActivity.java#L30)):*
<details><summary><i>Previously, AnmAtAnm (Andrew n marshall) wrote…</i></summary><blockquote>

unused?
</blockquote></details>

Done.

---

*[blocklylib-core/src/main/java/com/google/blockly/android/codegen/CodeGenerationRequest.java, line 44 at r6](https://reviewable.io:443/reviews/google/blockly-android/592#-KjshBmCxlXlbyQwTVcl:-Kjt0QyMa4SjavknUOeO:b-896fix) ([raw file](https://github.com/google/blockly-android/blob/739ce02273f11a19cf64e41a148b103c2a03427e/blocklylib-core/src/main/java/com/google/blockly/android/codegen/CodeGenerationRequest.java#L44)):*
<details><summary><i>Previously, AnmAtAnm (Andrew n marshall) wrote…</i></summary><blockquote>

Might as well mark these params @NotNull.
</blockquote></details>

Done.

---

*[blocklylib-core/src/main/java/com/google/blockly/android/codegen/LanguageDefinition.java, line 17 at r6](https://reviewable.io:443/reviews/google/blockly-android/592#-KjshzfZyTJ5lpU2c49J:-Kjt0uqAImAsqXcm9muw:b-896fix) ([raw file](https://github.com/google/blockly-android/blob/739ce02273f11a19cf64e41a148b103c2a03427e/blocklylib-core/src/main/java/com/google/blockly/android/codegen/LanguageDefinition.java#L17)):*
<details><summary><i>Previously, AnmAtAnm (Andrew n marshall) wrote…</i></summary><blockquote>

JavaDoc these public fields.
Also, mLanguageNamespace => mGeneratorRef, per offline.
</blockquote></details>

Done.

---

*[blocklydemo/src/main/java/com/google/blockly/android/demo/StylesActivity.java, line 22 at r5](https://reviewable.io:443/reviews/google/blockly-android/592#-KjsehoXn58OjPMK6sfR:-Kjt00t5GawCrCZ0EXtz:b-896fix) ([raw file](https://github.com/google/blockly-android/blob/fe40f53f19087814bb0a3f0ddf206e89cbc733b0/blocklydemo/src/main/java/com/google/blockly/android/demo/StylesActivity.java#L22)):*
<details><summary><i>Previously, AnmAtAnm (Andrew n marshall) wrote…</i></summary><blockquote>

unused?
</blockquote></details>

Done.

---

*[blocklydemo/src/main/java/com/google/blockly/android/demo/flyout/AlwaysOpenToolboxActivity.java, line 24 at r5](https://reviewable.io:443/reviews/google/blockly-android/592#-Kjsf0ljrbAeuf2PPvY8:-Kjt02EdyMSqMZqWsdE_:b-896fix) ([raw file](https://github.com/google/blockly-android/blob/fe40f53f19087814bb0a3f0ddf206e89cbc733b0/blocklydemo/src/main/java/com/google/blockly/android/demo/flyout/AlwaysOpenToolboxActivity.java#L24)):*
<details><summary><i>Previously, AnmAtAnm (Andrew n marshall) wrote…</i></summary><blockquote>

Unused?
</blockquote></details>

Done.

---

*[blocklydemo/src/main/java/com/google/blockly/android/demo/flyout/NoCategoriesToolboxActivity.java, line 11 at r5](https://reviewable.io:443/reviews/google/blockly-android/592#-Kjsf2WqOyOnghc_vi79:-Kjt03vJ0_x1sHfAhI2D:b-896fix) ([raw file](https://github.com/google/blockly-android/blob/fe40f53f19087814bb0a3f0ddf206e89cbc733b0/blocklydemo/src/main/java/com/google/blockly/android/demo/flyout/NoCategoriesToolboxActivity.java#L11)):*
<details><summary><i>Previously, AnmAtAnm (Andrew n marshall) wrote…</i></summary><blockquote>

Unused?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/google/blockly-android/592)*
<!-- Sent from Reviewable.io -->
